services/docker/docker-size.service.js

```diff
@@ -73,30 +114,76 @@ export default class DockerSize extends BaseJsonService {
     })
   }
 
-  transform({ tag, sort, data }) {
-    if (!tag && sort === 'date') {
-      if (data.count === 0) {
-        throw new NotFound({ prettyMessage: 'repository not found' })
+  noTagWithDateSortTransform(data, arch) {
+    if (data.count === 0) {
+      throw new NotFound({ prettyMessage: 'repository not found' })
+    } else {
+      const latestEntry = data.results[0]
+
+      if (arch) {
+        return { size: getImageForArch(latestEntry.images, arch) }
       } else {
-        return { size: data.results[0].full_size }
+        return { size: latestEntry.full_size }
       }
-    } else if (!tag && sort === 'semver') {
-      const [matches, versions] = data.reduce(
-        ([m, v], d) => {
-          m[d.name] = d.full_size
-          v.push(d.name)
-          return [m, v]
-        },
-        [{}, []]
-      )
-      const version = latest(versions)
+    }
+  }
+
+  noTagWithSemverSortTransform(data, arch) {
+    // If no tag is specified, and sorting is by semver, first filter out the entry containing the latest semver from the response with Docker images.
+    // Then check if any of the returned images for this entry has an architecture matching the arch parameter supplied by the user.
+    // If yes, return the size of the image with this arch.
+    // If not, return the value of the `full_size` from the entry matching the latest semver.
+
+    const [matches, versions, images] = data.reduce(
+      ([m, v, i], d) => {
+        m[d.name] = d.full_size
+        v.push(d.name)
+        i[d.name] = d.images
+        return [m, v, i]
+      },
+      [{}, [], {}]
+    )
+
+    const version = latest(versions)
+
+    let sizeOfImgWithArch
+
+    if (arch) {
+      Object.keys(images).forEach(ver => {
+        if (ver === version) {
+          sizeOfImgWithArch = getImageForArch(images[ver], arch)
+          return { size: sizeOfImgWithArch }
+        }
+      })
+      return { size: sizeOfImgWithArch }
```

Comment:
If we have iterated through every image and not returned anything inside the `forEach`, then this will `return { size: undefined }`. We don't want to do that. Instead, this line should throw an exception. Probably a `NotFound()`, or maybe an `InvalidResponse()` if we never expect to end up here.

Reply:
I modified it, now instead of 
```js 
return { size: sizeOfImgWithArch }
```
we have here:
```js
 if (sizeOfImgWithArch) {
        return { size: sizeOfImgWithArch }
      } else {
        throw new NotFound({ prettyMessage: 'architecture not found' })
      }
```